src/Emogrifier.php

```diff
@@ -1069,6 +1092,78 @@ private function attributeValueIsImportant($attributeValue)
         return strtolower(substr(trim($attributeValue), -10)) === '!important';
     }
 
+    /**
+     * Reconstructs the CSS containing only allowed @media rules from the "uninlineable" CSS rules array returned by
+     * `parseCssRules`.
+     *
+     * @param string[][] $cssRules
+     *
+     * @return string
+     */
+    private function combineCssMediaRules(array $cssRules)
```

Comment:
Can you also please split up this method? It's far too long.

Reply:
This is why I'd like to add a `CssConcatenator` class.

Reply:
Now using `CssConcatenator`.